packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -68,6 +68,12 @@ enum _ScaffoldSlot {
 /// [BuildContext] via [ScaffoldMessenger.of] and use the
 /// [ScaffoldMessengerState.showSnackBar] function.
 ///
+/// When the [ScaffoldMessenger] has nested [Scaffold] descendants, the
+/// ScaffoldMessenger will only present a [SnackBar] to the root Scaffold of
+/// the subtree of Scaffolds. In order to show SnackBars in the inner, nested
```

Comment:
It would be better to give code snippets and corresponding test cases！

Reply:
Here is my test code,
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(BugReport());
}

class BugReport extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: ScaffoldMessenger(
          child: Scaffold(
            body: Builder(
              builder: (context) => GestureDetector(
                onTap: () {
                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
                    content: const Text('inner'),
                    duration: const Duration(seconds: 1),
                    action: SnackBarAction(label: 'ACTION', onPressed: () {}),
                    behavior: SnackBarBehavior.floating,
                  ));
                },
                child: Container(color: Colors.purple,),
              ),
            ),
          ),
        ),
        floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
        floatingActionButton: Builder(
          builder: (context) => FloatingActionButton(onPressed: () {
            ScaffoldMessenger.of(context).showSnackBar(SnackBar(
              content: const Text('outer'),
              duration: const Duration(seconds: 5),
              action: SnackBarAction(label: 'ACTION', onPressed: () {}),
              behavior: SnackBarBehavior.floating,
            ));
          }),
        ),
      ),
    );
  }
}

```

Reply:
Thanks @xu-baolin ! 🎉